### PR TITLE
PED mailing list improvements

### LINKED ISSUE #128: [PE-D] Acceptance test for mailingList Command: error message recommand
Description: Although as you stated, invalid tag for mailingList will be ignored but for the irrelavent keywords typed in should not be ignored. This is actually a wrong command input.

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: HangZelin/ped#10

### LINKED ISSUE #162: [PE-D] Suggestion for cosmetic improvement for empty tag representation in .csv file from mailingList command
Exported csv file with "mailingList t/" on 3 persons with no tags.

Empty tags appear as "[]" in .csv file. Might be better to remove the brackets when exporting them, to be consistent with how other empty fields appear in the .csv files (like birthday)

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: hanif-kamal/ped#7

### PR DESCRIPTION
**Pull request by cookiedan42**
closes #128
closes #162

### CODE REVIEW

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 1]
```diff
@@ -202,14 +206,11 @@ private void handleMailingList() {
 
         List<Person> personList = logic.getFilteredPersonList();
         Set<Prefix> prefixSet = logic.getPrefixStore();
-
         try {
             CsvUtil.modelToCsv(personList, path, prefixSet);
         } catch (IOException e) {
-            logger.info("writing csv failed" + e.getMessage());
-            resultDisplay.setFeedbackToUser(e.getMessage());
+            throw new CommandException("writing csv failed: " + e.getMessage());
```
Comment (yucheng11122017) [inline src/main/java/seedu/address/ui/MainWindow.java:212]: Maybe capitalize w?

File: src/test/java/seedu/address/logic/parser/MailingListCommandParserTest.java [hunk 2]
```diff
@@ -23,4 +26,19 @@ public void parse_emptyArg_oneArg() {
...
+    public void parse_preambleExists_failure() {
+        Set<Prefix> prefixes = Set.of(PREFIX_PHONE);
+        MailingListCommand expectedCommand = new MailingListCommand(prefixes);
+        assertParseFailure(parser, "preamble " + PREFIX_PHONE.getPrefix(), MailingListCommand.MESSAGE_USAGE);
+    }
+
+    @Test
+    public void parse_extraDataExists_failure() {
+        Set<Prefix> prefixes = Set.of(PREFIX_PHONE);
+        MailingListCommand expectedCommand = new MailingListCommand(prefixes);
```
Comment (yucheng11122017) [inline src/test/java/seedu/address/logic/parser/MailingListCommandParserTest.java:40]: I think you don't need this expected command anymore? Unless I'm high...

File: src/test/java/seedu/address/logic/parser/MailingListCommandParserTest.java [hunk 3]
```diff
@@ -23,4 +26,19 @@ public void parse_emptyArg_oneArg() {
         MailingListCommand expectedCommand = new MailingListCommand(prefixes);
         assertParseSuccess(parser, " " + PREFIX_PHONE.getPrefix(), expectedCommand);
     }
+
+    @Test
+    public void parse_preambleExists_failure() {
+        Set<Prefix> prefixes = Set.of(PREFIX_PHONE);
+        MailingListCommand expectedCommand = new MailingListCommand(prefixes);
+        assertParseFailure(parser, "preamble " + PREFIX_PHONE.getPrefix(), MailingListCommand.MESSAGE_USAGE);
```
Comment (yucheng11122017) [inline src/test/java/seedu/address/logic/parser/MailingListCommandParserTest.java:34]: I think you also dunnid expectedCommand here?

**Review by yucheng11122017 — APPROVED** [review]
LGTM except for some small changes :)

**Review by leezhixuan — APPROVED** [review]
LGTM but your check style needs fixing

**Pull request merged**